theano/tensor/nlinalg.py

```diff
@@ -179,6 +185,10 @@ class ExtractDiag(Op):
     __props__ = ("view",)
 
     def __init__(self, view=False):
+        warnings.warn("DeprecationWarning: theano.tensor.nlinalg.ExtractDiag"
```

Comment:
Yes I saw it and that is what confuses me. How could it run on the GPU with only CPU code?